packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -337,7 +388,51 @@ class TextSelectionOverlay {
   /// {@macro flutter.widgets.SelectionOverlay.showToolbar}
   void showToolbar() {
     _updateSelectionOverlay();
-    _selectionOverlay.showToolbar();
+
+    if (selectionControls is! TextSelectionHandleControls) {
+      _selectionOverlay.showToolbar();
+      return;
+    }
+
+    if (contextMenuBuilder == null) {
+      return;
+    }
+
+    // If right clicking on desktop, use the right click position as the only
+    // anchor.
+    final RenderBox renderBox = context.findRenderObject()! as RenderBox;
+    if (renderObject.lastSecondaryTapDownPosition != null) {
+      switch (defaultTargetPlatform) {
```

Comment:
Since you have a context here, this should be `Theme.of(context).platform)`.

Reply:
Same comment above about the widgets library accessing Theme.